src/noisepy/seis/io/utils.py

```diff
@@ -67,6 +69,13 @@ def io_retry(func: Callable, *args, **kwargs) -> Any:
             time.sleep(FIND_RETRY_SLEEP * i)
 
 
+def date_range(month: int, start_day: int, end_day: int, start_hr: int = 0, end_hr: int = 0):
```

Comment:
this is also test-only code, so it just be with the tests, not the production code. I would keep it in tests/utils.py. If you need it in `noisepy-seis` you could duplicate it there. Why duplication is generally not great in this case it's the least of two evils and this is a very simple piece of test only code.